File: python/allergies/allergies.py
class Allergies:

    ALLERGENS = {
        'eggs': 1,
        'peanuts': 2,
        'shellfish': 4,
        'strawberries': 8,
        'tomatoes': 16,
        'chocolate': 32,
        'pollen': 64,
        'cats': 128,
    }

    def __init__(self, mask):
        self.mask = mask

    def allergic_to(self, item):
        return self.mask & self.ALLERGENS[item] > 0

    @property
    def lst(self):
        return [allergen for allergen in self.ALLERGENS if self.allergic_to(allergen)]

File: dart/allergies/lib/allergies.dart
class Allergies {
  static const allergens = const {
    "eggs": 1,
    "peanuts": 2,
    "shellfish": 4,
    "strawberries": 8,
    "tomatoes": 16,
    "chocolate": 32,
    "pollen": 64,
    "cats": 128
  };

  bool allergicTo(String aliment, int mask) => mask & allergens[aliment]! != 0;

  List<String> list(int mask) =>
      allergens.keys.where((n) => allergicTo(n, mask)).toList();


}
